cli/cluster/cmd/replicaset.go

```diff
@@ -268,3 +274,61 @@ func Expel(uri *url.URL, ctx ExpelCtx) error {
 	}
 	return err
 }
+
+// RolesAddCtx describes the context to add role of instance.
+type RolesAddCtx struct {
+	// InstName is an instance name in which add or remove role.
+	InstName string
+	// GroupName is an replicaset name in which add or remove role.
+	GroupName string
+	// ReplicasetName is an replicaset name in which add or remove role.
+	ReplicasetName string
+	// IsGlobal is an boolean value if role needs to add in global scope.
+	IsGlobal bool
+	// RoleName is a name of role to add.
+	RoleName string
+	// Publishers is data publisher factory.
+	Publishers libcluster.DataPublisherFactory
+	// Collectors is data collector factory.
+	Collectors libcluster.DataCollectorFactory
+	// Username defines a username for connection.
+	Username string
+	// Password defines a password for connection.
+	Password string
+	// Force true if the key selection for patching the config
+	// should be skipped.
+	Force bool
+}
+
+// AddRole adds a role by patching the cluster config.
+func AddRole(uri *url.URL, ctx RolesAddCtx) error {
+	opts, err := ParseUriOpts(uri)
+	if err != nil {
+		return fmt.Errorf("invalid URL %q: %w", uri, err)
+	}
+	connOpts := connectOpts{
+		Username: ctx.Username,
+		Password: ctx.Password,
+	}
+
+	collector, publisher, closeFunc, err := createDataCollectorAndKeyPublisher(
+		ctx.Collectors, ctx.Publishers, opts, connOpts)
+	if err != nil {
+		return err
+	}
+	defer closeFunc()
+	source := replicaset.NewCConfigSource(collector, publisher,
```

Comment:
Just to separate code blocks.
```suggestion
	defer closeFunc()

	source := replicaset.NewCConfigSource(collector, publisher,
```

Reply:
Added blank line.